File: backend/app/routers/chunks.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.db import get_db
from app.models import Chunk

router = APIRouter(prefix="/api/chunks", tags=["chunks"])


@router.get("/{chunk_id}")
def get_chunk(chunk_id: int, db: Session = Depends(get_db)):
    chunk = db.get(Chunk, chunk_id)
    if chunk is None:
        raise HTTPException(status_code=404, detail="Chunk not found")
    return {
        "chunk_id": chunk.id,
        "document_id": chunk.document_id,
        "filename": chunk.document.original_filename,
        "pdf_url": f"/api/documents/{chunk.document_id}/pdf",
        "page_number": chunk.page_number,
        "bboxes": chunk.bboxes,
        "text": chunk.text,
        "context_header": chunk.context_header,
    }

File: frontend/src/api/chunks.ts
import { useQuery } from "@tanstack/react-query";

import { apiFetch } from "./client";

export interface BoundingBox {
  page_width: number;
  page_height: number;
  rects: { x0: number; y0: number; x1: number; y1: number }[];
}

export interface ChunkDetail {
  chunk_id: number;
  document_id: number;
  filename: string;
  pdf_url: string;
  page_number: number;
  bboxes: BoundingBox;
  text: string;
  context_header: string | null;
}

export function useChunkDetail(chunkId: number | null) {
  return useQuery({
    queryKey: ["chunk", chunkId],
    queryFn: () => apiFetch<ChunkDetail>(`/api/chunks/${chunkId}`),
    enabled: chunkId !== null,
  });
}
